orderer/consensus/hlmirbft/chain.go

```diff
@@ -876,38 +891,96 @@ func (c *Chain) writeBlock(block *common.Block, index uint64) {
 //   -- pending bool; if there are envelopes pending to be ordered,
 //   -- err error; the error encountered, if any.
 // It takes care of config messages as well as the revalidation of messages if the config sequence has advanced.
-func (c *Chain) ordered(msg *orderer.SubmitRequest) (batches [][]*common.Envelope, pending bool, err error) {
+
+//FLY2-64 PROPOSED CHANGE
+// - from ordered we propose the message
+func (c *Chain) ordered(msg *orderer.SubmitRequest) (err error) {
 	seq := c.support.Sequence()
 
 	if c.isConfig(msg.Payload) {
-		// ConfigMsg
+		//FLY2-57 proposed change
+		// - proposing the configMsg
+		c.configInflight = true
+
 		if msg.LastValidationSeq < seq {
 			c.logger.Warnf("Config message was validated against %d, although current config seq has advanced (%d)", msg.LastValidationSeq, seq)
-			msg.Payload, _, err = c.support.ProcessConfigMsg(msg.Payload)
-			if err != nil {
-				c.Metrics.ProposalFailures.Add(1)
-				return nil, true, errors.Errorf("bad config message: %s", err)
+			if err := c.proposeMsg(msg); err != nil {
+				return err
 			}
 		}
 
-		batch := c.support.BlockCutter().Cut()
-		batches = [][]*common.Envelope{}
-		if len(batch) != 0 {
-			batches = append(batches, batch)
-		}
-		batches = append(batches, []*common.Envelope{msg.Payload})
-		return batches, false, nil
-	}
-	// it is a normal message
-	if msg.LastValidationSeq < seq {
+		// batch := c.support.BlockCutter().Cut()
+		// batches = [][]*common.Envelope{}
+		// if len(batch) != 0 {
+		// 	batches = append(batches, batch)
+		// }
+		// batches = append(batches, []*common.Envelope{msg.Payload})
+		// return batches, false, nil
+	} else if msg.LastValidationSeq < seq { // it is a normal message
 		c.logger.Warnf("Normal message was validated against %d, although current config seq has advanced (%d)", msg.LastValidationSeq, seq)
 		if _, err := c.support.ProcessNormalMsg(msg.Payload); err != nil {
 			c.Metrics.ProposalFailures.Add(1)
-			return nil, true, errors.Errorf("bad normal message: %s", err)
+			return errors.Errorf("bad normal message: %s", err)
+		}
+
+		//FLY2-64 - proposed change
+		// new function to propose message
+		if err := c.proposeMsg(msg); err != nil {
+			return err
 		}
 	}
-	batches, pending = c.support.BlockCutter().Ordered(msg.Payload)
```

Comment:
As the proposing of the message occurs at the end of both if statements can just have a single call after both checks